pkg/handler3/handler.go

```diff
@@ -96,10 +122,17 @@ func (o *OpenAPIService) getGroupBytes() ([]byte, error) {
 	}
 
 	sort.Strings(keys)
-	group := make(map[string][]string)
-	group["Paths"] = keys
-
-	j, err := json.Marshal(group)
+	discovery := &OpenAPIV3Discovery{Paths: make(map[string]OpenAPIV3DiscoveryGroupVersion)}
+	for gvString, groupVersion := range o.v3Schema {
+		etagBytes, err := groupVersion.etagCache.Get()
```

Comment:
Maybe add a TODO here (or inside the etagCache), this is forcing a json marshal of all group/versions, which is unfortunately going to be slow and we would like to replace that with a faster hashing mechanism.